cpp/src/distance/fused_distance_nn-ext.cuh

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "fused_distance_nn_helpers.cuh"  // include initialize and reduce operations
+#include <raft/core/kvp.hpp>              // raft::KeyValuePair
+#include <raft/core/resources.hpp>        // raft::resources
+#include <raft/util/raft_explicit.hpp>    // RAFT_EXPLICIT
+
+#include <cstdint>  // int64_t
+
+#ifdef RAFT_EXPLICIT_INSTANTIATE_ONLY
```

Comment:
Do we define this macro in cuvs? Why are we keeping the -ext and -inl structure?

Reply:
This is just to get the code over and a public API exposed. We can and should continue to clean these up over time. Main concern right now is clean public api, proper source file etiquette, and build time that's not horrendous.

Reply:
One thing I probably should do is search and replace for all the template instantiation macros that have _raft_ in the name. Don't need to keep those raft.

Reply:
What we should do right now is just keep the -inl file then and call it a day. It will work as-is without needing any of the extra fluff that comes in the -ext and the main file, and we also won't have to worry about macros.